src/ResourceManager/Common/Commands.ResourceManager.Common/ResourceGroupCompleter.cs

```diff
@@ -0,0 +1,134 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Common.TabCompletion
+{
+    using Commands.Common.Authentication;
+    using Commands.Common.Authentication.Abstractions;
+    using Management.Internal.Resources;
+    using Management.Internal.Resources.Models;
+    using Properties;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    /// <summary>
+    /// This attribute will allow the user to autocomplete the -ResourceGroup parameter of a cmdlet with valid resource groups
+    /// </summary>
+    public class ResourceGroupCompleterAttribute : ArgumentCompleterAttribute
+    {
+        private static IList<String> _resourceGroupNames = new List<string>();
+        private static readonly object _lock = new object();
+
+        protected static IList<String> ResourceGroupNames
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    _resourceGroupNames = new List<string>();
+                    IAzureContext context = AzureRmProfileProvider.Instance.Profile.DefaultContext;
+                    try
+                    {
+                        var instance = AzureSession.Instance;
+                        var client = instance.ClientFactory.CreateCustomArmClient<ResourceManagementClient>(
```

Comment:
Use the simpler CreateArmClient method here: https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common.Authentication.Abstractions/Interfaces/IClientFactory.cs#L39

Reply:
Staying the same as discussed offline